packages/react-select/src/Select.tsx

```diff
@@ -764,7 +764,8 @@ export default class Select<
 
     let newAriaSelection = ariaSelection;
 
-    let hasKeptFocus = isFocused && prevWasFocused;
+    let hasKeptFocus =
```

Comment:
Ensure that since we are setting focus back to select input after clearing the value, initial-input-focus is not called again.